src/components/Form/FormUpload.js

```diff
@@ -0,0 +1,118 @@
+import React, { useState } from 'react'
+import firebase from '../../utils/Firebase'
+import { v4 as uuidv4 } from 'uuid'
+
+//third party components
+import { useFormikContext } from "formik";
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+
+
+//custom components
+import FieldError from "../Error/FieldError"
+
+const storage = firebase.storage().ref()
+
+
+const FormUpload = ({ name, icon = false, disabled = false, hidden = false, sizeLimit, fileDestination, fileType }) => {
+
+      let { values, errors } = useFormikContext()
+      const { touched } = useFormikContext()
+   
+    const [progress, setProgress] = useState(0)
+    const [showError, setShowError] = useState(false)
+
+    const getUploadError = (isError, err) => {
+
+        values[name] = ""
+
+        if (isError) {
+            setShowError(true)
+            errors[name] = err
+        }else {
+            setShowError(false)
+            errors[name] = false
+        }
+    }
+
+ 
+
+    const onChange = () => {
+
+        let file = document.querySelector(`#uploadFile`).files[0]
```

Comment:
This isn't a good way to get the file data.

Reply:
Let your `onChange` function take in the `event` object and you can get the file data by doing
```
let file = event.target.files[0]
````

Reply:
After approving a pending transaction, the data was approved and ever since then if I load the page(or refresh the page) the state is not being populated appropriately, only two out of all the processed transaction is being sent to the state until I click on a modal.

Reply:
Will work on the upload file stuff.

On your second comment....yeah, same thing happens to me... got no idea why it does that. I've debugged it and all data is sent back up until time to updating the store. but for some reason redux waits on some until you try clicking a modal.